bundles/admin/admin-announcements/view/AnnouncementsList.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { AnnouncementsForm } from './AnnouncementsForm';
+import PropTypes from 'prop-types';
+import { Collapse } from "antd";
+import { PlusOutlined } from '@ant-design/icons';
+import { Button } from 'oskari-ui';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import styled from 'styled-components';
+
+/*
+Maps all created announcements into a collapse.
+*/
+
+const { Panel } = Collapse;
+
+const StyledButton = styled(Button)`
+    margin-top: 10px;
+`;
+const AnnouncementsList = ({controller,  announcements, updated, title, activeKey }) => {
+
+  // TODO(?) Create maybe a better way to handle updating the collapse panels
+  if (!updated) {
+    controller.getAdminAnnouncements(function(ann){
+      var announcements = [];
+
+        ann.data.forEach((announcement) => {
+          announcements.push(announcement);
+        });
+        controller.pushAnnouncements(announcements);
+      });
+  }
```

Comment:
Check the suggestion about triggering the fetch when the handler is initialized. With that change you should be able to remove this (and the pushAnnouncements() from handler).